test/integration_test/snapshot_test.go

```diff
@@ -1180,6 +1187,479 @@ func updateRetainValueTest(t *testing.T) {
 	log.InfoD("Test status at end of %s test: %s", t.Name(), testResult)
 }
 
+// deleteErrorSnapshotCRTest tests the workflow in which stork deletes the
+// snapshot CRs of snapshots that are in Error state and older than a certain
+// defined time period.
+// 1. Start a volumesnapshot and take one successful snapshot.
+// 2. Bring down the node where the PVC is attached.
+// 3. Wait for couple of snapshots to go into `Error` state.
+// 4. Check if they get cleaned up after 90 seconds.
+// 5. Check if the snapshotschedule object gets updated for the deleted snapshot entries.
+func deleteErrorSnapshotCRTest(t *testing.T) {
+	var testrailID, testResult = 301162, testResultFail
+	runID := testrailSetupForTest(testrailID, &testResult, t.Name())
+	defer updateTestRail(&testResult, testrailID, runID)
+	defer updateDashStats(t.Name(), &testResult)
+
+	// Deploy a dummy application for taking snapshots.
+	ctx := createApp(t, "delete-error-snapshot-cr-test")
+
+	////////////////////////////////////////////////////
+	// Create schedulepolicy and snapshot schedule.
+	////////////////////////////////////////////////////
+	policyName := "intervalpolicy"
+	scheduleName := "errsnap-schedule"
+	retain := 5
```

Comment:
Lets make the retain value small like 3, and then lets have in this same test check
1. When first snapshot itself errors out
2. When first 2 snasphots are successful and then the third one errors out
3. When first 3 snaps are successful and the 4th snapshot errors out.

By doing this we can cover allmost all the cases for the schedule functionality wise.

Reply:
Can you please elaborate what is meant to be tested here?

Reply:
I was thinking of handling all the 3 cases with error snapshots in integration test. Anyway we can take it later in another PR.